.github/workflows/master.yml

```diff
@@ -327,6 +327,7 @@ jobs:
       run_command: |
         python3 build_report_check.py "$CHECK_NAME"
   MarkReleaseReady:
+    if: ${{ ! (contains(needs.*.result, 'skipped') || contains(needs.*.result, 'failure')) }}
```

Comment:
${{ !failure() && !cancelled() }} should work

Reply:
it's either broken in https://github.com/ClickHouse/ClickHouse/actions/runs/7896315258/job/21575897913 or in https://github.com/ClickHouse/ClickHouse/actions/runs/7904523295/job/21577047648

We should explicitly check that every needed job succeeded. Not the tree of all dependencies